services/pull/merge.go

```diff
@@ -34,8 +34,8 @@ import (
 	issue_service "code.gitea.io/gitea/services/issue"
 )
 
-// GetDefaultMergeMessage returns default message used when merging pull request
-func GetDefaultMergeMessage(ctx context.Context, baseGitRepo *git.Repository, pr *issues_model.PullRequest, mergeStyle repo_model.MergeStyle) (message, body string, err error) {
+// Compose message used when merging pull request.
+func getMergeMessage(ctx context.Context, baseGitRepo *git.Repository, pr *issues_model.PullRequest, mergeStyle repo_model.MergeStyle, commitTitle, commitBody string) (message, body string, err error) {
```

Comment:
What about use `extraVars map[string]string` instead of `commitTitle, commitBody string`? Then merge the `extraVars` into `vars`. It seems more flexible, and no need to do `if mergeStyle == repo_model.MergeStyleRebase` check.